noir-projects/noir-contracts/contracts/key_registry_contract/src/main.nr

```diff
@@ -0,0 +1,119 @@
+contract KeyRegistry {
+  use dep::std::hash::poseidon2::Poseidon2::hash as poseidon2_hash;
+  use dep::aztec::{
+    context::gas::GasOpts,
+    state_vars::{
+      SharedMutable,
+      Map
+    }, 
+    protocol_types::{
+      abis::function_selector::FunctionSelector,
+      contract_class_id::ContractClassId,
+      address::{
+        AztecAddress,
+        EthAddress,
+        PublicKeysHash,
+        PartialAddress,
+      },
+      constants::{
+        GENERATOR_INDEX__CONTRACT_ADDRESS_V1,
+        GENERATOR_INDEX__PUBLIC_KEYS_HASH
+      },
+      traits::{
+        Serialize,
+        Deserialize,
+      }
+    },
+  };
+
+  global KEY_ROTATION_DELAY = 5;
+
+  struct PublicKeys {
```

Comment:
This struct is such an unnecessary complexity. You could nuke half the file while keeping the functionality.